launchflow/aws/lambda_service.py

```diff
@@ -1,88 +1,264 @@
+from launchflow.service import ServiceOutputs
+from launchflow.workflows.utils import zip_source
+
+
+def _get_relative_handler_import_path(func: Callable) -> str:
+    # Get the absolute path of the directory containing the launchflow.yaml file
+    launchflow_yaml_abspath = os.path.dirname(
+        os.path.abspath(config.launchflow_yaml.config_path)
+    )
+    # Get the absolute path of the file where the function is defined
+    func_file_path = os.path.abspath(inspect.getfile(func))
+    # Ensure the cwd has a trailing slash to avoid issues with relative path calculation
+    cwd = os.path.abspath(launchflow_yaml_abspath) + os.path.sep
+    # Convert the absolute path to a relative path with respect to the current working directory
+    relative_path = os.path.relpath(func_file_path, cwd)
+    # Convert the relative file path to a Python module path
+    module_path = relative_path.replace(os.path.sep, ".").rsplit(".py", 1)[0]
+    # Append the function name to the module path
+    full_import_path = f"{module_path}.{func.__name__}"
+
+    return full_import_path
+
+
+def _clean_pycache(directory: str):
+    for root, dirs, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".pyc"):
+                os.remove(os.path.join(root, file))
+        for dir in dirs:
+            if dir == "__pycache__":
+                shutil.rmtree(os.path.join(root, dir))
+
+
+def _zip_source(
+    build_directory: str,
+    build_ignore: List[str],
+    python_version: Union[str, None],
+    requirements_txt_path: Optional[str],
+):
+    # 1. create a temp dir
+    with tempfile.TemporaryDirectory() as temp_dir:
+        shutil.copytree(build_directory, temp_dir, dirs_exist_ok=True, symlinks=True)
+
+        # 2. Install packages from requirements.txt (if specified)
+        if requirements_txt_path is not None and python_version is not None:
+            # TODO: Update this to use uv for faster builds
+            subprocess.check_call(
+                f"pip install --no-cache-dir --platform manylinux2014_x86_64 --target={temp_dir} --implementation cp --python-version {python_version} --only-binary=:all: -r {requirements_txt_path}".split(),
+                cwd=config.launchflow_yaml.project_directory_abs_path,
+                stdout=subprocess.DEVNULL,  # TODO: Dump to the build logs file
+                stderr=subprocess.DEVNULL,  # TODO: Dump to the build logs file
+            )
+            _clean_pycache(temp_dir)
+
+        # 3. Zip the contents of the temp directory
+        zip_file_path = os.path.join(temp_dir, "lambda.zip")
+        zip_source(temp_dir, ignore_patterns=build_ignore, file=zip_file_path)
+
+        # Read the zip file
+        with open(zip_file_path, "rb") as zip_file:
+            zip_file_content = zip_file.read()
+
+    return zip_file_content
 
 
 @dataclass
 class LambdaServiceInputs(Inputs):
     pass
 
 
-class LambdaStaticService(AWSStaticService):
+@dataclass
+class PythonRuntime:
```

Comment:
nit: add doc comments so this shows up in the docs

In general anything that is an input to a resource should be documented

Reply:
done